crates/cheatcodes/src/defs/vm.rs

```diff
@@ -0,0 +1,1117 @@
+
+    /// Derives a private key from the name, labels the account with that name, and returns the wallet.
+    #[cheatcode(group = Utilities)]
+    function createWallet(string calldata walletLabel) external returns (Wallet memory wallet);
+
+    /// Generates a wallet from the private key and returns the wallet.
+    #[cheatcode(group = Utilities)]
+    function createWallet(uint256 privateKey) external returns (Wallet memory wallet);
+
+    /// Generates a wallet from the private key, labels the account with that name, and returns the wallet.
+    #[cheatcode(group = Utilities)]
+    function createWallet(uint256 privateKey, string calldata walletLabel) external returns (Wallet memory wallet);
+
+    /// Get a `Wallet`'s nonce.
+    #[cheatcode(group = Utilities)]
+    function getNonce(Wallet calldata wallet) external returns (uint64 nonce);
+
+    /// Signs data with a `Wallet`.
+    #[cheatcode(group = Utilities)]
+    function sign(Wallet calldata wallet, bytes32 digest) external returns (uint8 v, bytes32 r, bytes32 s);
+
+    /// Derive a private key from a provided mnenomic string (or mnenomic file path)
+    /// at the derivation path `m/44'/60'/0'/0/{index}`.
+    #[cheatcode(group = Utilities)]
+    function deriveKey(string calldata mnemonic, uint32 index) external pure returns (uint256 privateKey);
+    /// Derive a private key from a provided mnenomic string (or mnenomic file path)
+    /// at `{derivationPath}{index}`.
+    #[cheatcode(group = Utilities)]
+    function deriveKey(string calldata mnemonic, string calldata derivationPath, uint32 index)
+        external
+        pure
+        returns (uint256 privateKey);
+    /// Derive a private key from a provided mnenomic string (or mnenomic file path) in the specified language
+    /// at the derivation path `m/44'/60'/0'/0/{index}`.
+    #[cheatcode(group = Utilities)]
+    function deriveKey(string calldata mnemonic, uint32 index, string calldata language)
+        external
+        pure
+        returns (uint256 privateKey);
+    /// Derive a private key from a provided mnenomic string (or mnenomic file path) in the specified language
+    /// at `{derivationPath}{index}`.
+    #[cheatcode(group = Utilities)]
+    function deriveKey(string calldata mnemonic, string derivationPath, uint32 index, string calldata language)
+        external
+        pure
+        returns (uint256 privateKey);
+
+    /// Adds a private key to the local forge wallet and returns the address.
+    #[cheatcode(group = Utilities)]
+    function rememberKey(uint256 privateKey) external returns (address keyAddr);
+
+    // -------- Uncategorized Utilities --------
+
+    /// Labels an address in call traces.
+    #[cheatcode(group = Utilities)]
+    function label(address account, string calldata newLabel) external;
+
+    /// Gets the label for the specified address.
+    #[cheatcode(group = Utilities)]
+    function getLabel(address account) external returns (string memory currentLabel);
+}
```

Comment:
i love this usage of the sol! macro showcasing derive macro composition with the cheatcode macro, beautiful. had not appreciated how mayn cheats we have now